lib/api/routes/iterations.rb

```diff
@@ -14,10 +14,16 @@ class Iterations < Core
           halt 401, {error: "Please double-check your exercism API key."}.to_json
         end
 
-        if UserExercise.new(user_id: current_user.id, language: language, slug: slug, iteration_count: 0, state: 'unstarted').save
+        exercise = UserExercise.where(user_id: current_user.id, language: language, slug: slug).first_or_initialize(iteration_count: 0, state: 'unstarted')
```

Comment:
believe you can use `first_or_create!` so you don't have to do the `new_record?` check